backend/coins/btc/handlers/handlers.go

```diff
@@ -481,11 +482,16 @@ func (handlers *Handlers) postVerifyAddress(r *http.Request) (interface{}, error
 }
 
 func (handlers *Handlers) getCanVerifyExtendedPublicKey(_ *http.Request) (interface{}, error) {
-	btcAccount, ok := handlers.account.(*btc.Account)
-	if !ok {
-		return nil, errp.New("An account must be BTC based to support xpub verification")
+	switch specificAccount := handlers.account.(type) {
+	case *btc.Account:
+		return specificAccount.CanVerifyExtendedPublicKey(), nil
+	case *eth.Account:
+		// TODO: implement secure verification for ETH addresses
+		return nil, nil
```

Comment:
doesn't this produce an error in the info screen in eth (`null.includes(index)`?).

In any case, for non btc accounts you can just return an empty slice. (no panic in default)

Reply:
Doesn't produce an error since `null.includes(index)` is never reached when there is an address and there always has to be either an address or an xpub. 
But you're right anyway, returning an empty slice instead is better for consistency.